packages/pn-personagiuridica-webapp/src/pages/ApiIntegration.page.tsx

```diff
@@ -8,20 +8,33 @@ import PublicKeys from '../components/IntegrazioneApi/PublicKeys';
 import VirtualKeys from '../components/IntegrazioneApi/VirtualKeys';
 import LoadingPageWrapper from '../components/LoadingPageWrapper/LoadingPageWrapper';
 import { PNRole } from '../redux/auth/types';
-import { useAppSelector } from '../redux/hooks';
+import { checkPublicKeyIssuer } from '../redux/apikeys/actions';
+import { useAppDispatch, useAppSelector } from '../redux/hooks';
 import { RootState } from '../redux/store';
-import { PublicKeyStatus } from '../generated-client/pg-apikeys';
+import { PublicKeysIssuerResponseIssuerStatusEnum, PublicKeyStatus } from '../generated-client/pg-apikeys';
+import IntegrationApiBanner from '../components/IntegrazioneApi/IntegrationApiBanner';
 
 const ApiIntegration: React.FC = () => {
+  const dispatch = useAppDispatch();
   const { t } = useTranslation(['integrazioneApi']);
   const currentUser = useAppSelector((state: RootState) => state.userState.user);
   const role = currentUser.organization?.roles ? currentUser.organization?.roles[0] : null;
   const userHasAdminPermissions = useHasPermissions(role ? [role.role] : [], [PNRole.ADMIN]);
   const publicKeys = useAppSelector((state: RootState) => state.apiKeysState.publicKeys);
-  const hasValidKey =publicKeys.items.some((key) => key.status === PublicKeyStatus.Active ||  key.status === PublicKeyStatus.Rotated);
+  const hasPublicActive = !!publicKeys.items.find(el=>el.status === PublicKeyStatus.Active);
+  const keyIssuer = useAppSelector((state: RootState) => state.apiKeysState.issuerState);
 
+  const fetchCheckIssuer = useCallback(() => {
+    void dispatch(checkPublicKeyIssuer())
+      .unwrap();
+  }, []); 
+
+  useEffect(()=>{
+    fetchCheckIssuer();
+  },[]);
   
   const isAdminWithoutGroups = userHasAdminPermissions && !currentUser.hasGroup;
+  const integrationApiIsEnabled = isAdminWithoutGroups ? keyIssuer.issuer.issuerStatus === PublicKeysIssuerResponseIssuerStatusEnum.Active && hasPublicActive : keyIssuer.issuer.issuerStatus === PublicKeysIssuerResponseIssuerStatusEnum.Active;
```

Comment:
I have a couple of questions:
1. do we need the hasPublicActive check? the keyIssuer.issuer.issuerStatus is not enough?
2. I don't understand what integrationApiIsEnabled means. What we are checking here?